src/Build.UnitTests/BuildEventArgsSerialization_Tests.cs

```diff
@@ -892,8 +892,10 @@ public void ReadingCorruptedStreamThrows()
                 memoryStream.SetLength(i); // pretend that the stream abruptly ends
                 memoryStream.Position = 0;
 
+#pragma warning disable CA2000 // The memory stream needs to keep the binary reader open.
```

Comment:
nit: `BuildEventArgsReader` can be instructed to leave the reader open by setting `CloseInput = false`.

Reply:
The default CloseInput is false https://github.com/dotnet/msbuild/blob/9bea8026aad964cb36f3ec9d93bd95a941487690/src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs#L79-L82